src/main/java/docSharing/service/DocumentService.java

```diff
@@ -25,6 +35,45 @@ private DocumentService(DocumentRepository documentRepository, FolderRepository
         this.folderRepository = folderRepository;
     }
 
+    static Map<Integer,String> documentCacheChanges = new HashMap<>();//doc id, doc content
+    static Map<Integer,String> documentCacheDBContent = new HashMap<>();
+
+    public void update(int documentId, UpdateRequest updateRequest) {
+        Document document = documentRepository.getReferenceById(documentId);
+        document.updateContent(updateRequest);
+
+        updateContentOnCache(documentId, document.getContent().getContent());
+    }
+
+    private void updateContentOnCache(int documentId, String content){
+        documentCacheChanges.put(documentId,content);
+    }
+
+    private void deleteFromCache(int documentID){
+        documentCacheChanges.remove(documentID);
+        documentCacheDBContent.remove(documentID);
+    }
+
+    @Scheduled(fixedDelay = 10000) //10 seconds
+    private void updateContentOnDB(){
+        documentCacheChanges.forEach((key, value)->{
+            if (!documentCacheDBContent.containsKey(key) || !value.equals(documentCacheDBContent.get(key))) {
+                SaveToDB(key);
```

Comment:
instead of saveToDB:
document = documentRepository.getReferenceById(documentId);
document.setContent(value);
documentRepository.save(document);
documentCacheDBContent.put(key, value);